code/modules/mob/living/simple_animal/abnormality/aleph/seasons.dm

```diff
@@ -0,0 +1,779 @@
+	current_weather = new_weather //legacy code
+
+/mob/living/simple_animal/hostile/abnormality/seasons/proc/EndWeather()
+	if(current_weather) //legacy code
+		SSweather.end_weather(current_weather)
+
+//Combat
+/mob/living/simple_animal/hostile/abnormality/seasons/AttackingTarget(atom/attacked_target)
+	if(!can_act)
+		return FALSE
+	if((cone_attack_cooldown <= world.time) && prob(35))
+		return ConeAttack(target)
+	if((slam_cooldown <= world.time) && prob(35))
+		return Slam()
+	if(ishuman(target))
+		if(Finisher(target))
+			return
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/seasons/OpenFire()
+	if(!can_act)
+		return
+	if(cone_attack_cooldown <= world.time)
+		ConeAttack(target)
+		return
+	if((slam_cooldown <= world.time))
+		Slam()
+		return
+	..()
+
+/mob/living/simple_animal/hostile/abnormality/seasons/Life()
+	. = ..()
+	if(downgraded)
+		return
+	if(src.IsContained())
+		return
+	if((pulse_cooldown <= world.time) && prob(35))
+		return Pulse()
+
+/mob/living/simple_animal/hostile/abnormality/seasons/Move()
+	if(!can_act)
+		return FALSE
+	var/list/spawn_area = range(1, src)
+	for(var/turf/open/O in spawn_area)
+		if(!isturf(O) || isspaceturf(O))
+			continue
+		if(locate(/obj/effect/season_turf/temporary) in O)
+			continue
+		new /obj/effect/season_turf/temporary(O)
+	..()
+
+
+/mob/living/simple_animal/hostile/abnormality/seasons/proc/ConeAttack(atom/at = target)
+	cone_attack_cooldown = world.time + cone_attack_cooldown_time
+	playsound(get_turf(src), 'sound/abnormalities/seasons/breath_attack.ogg', 40, FALSE, 5)
+	SLEEP_CHECK_DEATH(1 SECONDS)
+	var/range = 15
+	var/list/turfs = list()
+	turfs = LineTarget(-40, range, at)
+	INVOKE_ASYNC(src, .proc/FireLine, src, turfs)
+	turfs = LineTarget(-20, range, at)
+	INVOKE_ASYNC(src, .proc/FireLine, src, turfs)
+	turfs = LineTarget(0, range, at)
+	INVOKE_ASYNC(src, .proc/FireLine, src, turfs)
+	turfs = LineTarget(20, range, at)
+	INVOKE_ASYNC(src, .proc/FireLine, src, turfs)
+	turfs = LineTarget(40, range, at)
+	INVOKE_ASYNC(src, .proc/FireLine, src, turfs)
+
+/mob/living/simple_animal/hostile/abnormality/seasons/proc/LineTarget(offset, range, atom/at = target)
+	if(!at)
+		return
+	var/turf/T = get_ranged_target_turf_direct(src, at, range, offset)
+	return (getline(src, T) - get_turf(src))
+
+/mob/living/simple_animal/hostile/abnormality/seasons/proc/FireLine(atom/source, list/turfs)
+	var/attacktype = breaching_stats[current_season][5]
+	var/list/hit_list = list()
+	for(var/turf/T in turfs)
+		if(istype(T, /turf/closed))
+			break
+		new attacktype(T)
+		for(var/mob/living/L in T.contents)
+			if(L in hit_list || istype(L, source.type))
+				continue
+			hit_list += L
+			L.apply_damage(cone_attack_damage, melee_damage_type, null, L.run_armor_check(null, melee_damage_type), spread_damage = TRUE)
+			to_chat(L, "<span class='userdanger'>You have been hit by [source]'s breath attack!</span>")
+			if(ishuman(L))
+				Finisher(L)
+		SLEEP_CHECK_DEATH(1)
```

Comment:
```suggestion
	FireLine(turfs)
	turfs = LineTarget(-20, range, at)
	FireLine(turfs)
	turfs = LineTarget(0, range, at)
	FireLine(turfs)
	turfs = LineTarget(20, range, at)
	FireLine(turfs)
	turfs = LineTarget(40, range, at)
	FireLine(turfs)

/mob/living/simple_animal/hostile/abnormality/seasons/proc/LineTarget(offset, range, atom/at = target)
	set waitfor = FALSE
	if(!at)
		return
	var/turf/T = get_ranged_target_turf_direct(src, at, range, offset)
	return (getline(src, T) - get_turf(src))

/mob/living/simple_animal/hostile/abnormality/seasons/proc/FireLine(list/turfs)
	set waitfor = FALSE
	var/attacktype = breaching_stats[current_season][5]
	var/list/hit_list = list()
	for(var/turf/T in turfs)
		if(istype(T, /turf/closed))
			break
		new attacktype(T)
		for(var/mob/living/L in T.contents)
			if(L in hit_list || istype(L, type))
				continue
			hit_list += L
			L.apply_damage(cone_attack_damage, melee_damage_type, null, L.run_armor_check(null, melee_damage_type), spread_damage = TRUE)
			to_chat(L, "<span class='userdanger'>You have been hit by [src]'s breath attack!</span>")
			if(ishuman(L))
				Finisher(L)
		SLEEP_CHECK_DEATH(1)
```

Reply:
`set waitfor = FALSE` should be used over INVOKE_ASYNC in cases where INVOKE_ASYNC is the only way they're meant to be called, as INVOKE_ASYNC more or less runs the proc with `set waitfor = FALSE`